### PR TITLE
[CodingStyle] Skip CatchExceptionNameMatchingTypeRector on variable previously defined

### LINKED ISSUE #4946: CatchExceptionNameMatchingTypeRector overrides variable
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | dev-master            |
| Installed as   | composer dependency                    |

CatchExceptionNameMatchingTypeRector renames exception variable event when it already exists. Which effectively changes code behavior and overrides variable.

This happens usually when there are nested try..catch blocks.

## Minimal PHP Code Causing Issue

https://getrector.org/demo/410f8a1c-9113-4f4f-895a-55e4305f9dc5#demo_result

## Expected Behaviour

To NOT renames variables like this OR generate some unique name. I would prefer no do renaming only when name is short.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #4946

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/Rector/AbstractRector/BetterStandardPrinterTrait.php [hunk 1]
```diff
@@ -78,4 +79,30 @@ protected function areNodesEqual($firstNode, $secondNode): bool
...
+            $previous = $previous->getAttribute(AttributeKey::PREVIOUS_NODE);
+        }
+
+        $parent = $node->getAttribute(AttributeKey::PARENT_NODE);
+        if (! $parent instanceof Node) {
+            return false;
+        }
+
+        return $this->isNodeFoundInPrevious($parent, $find);
+    }
```
Comment (TomasVotruba) [inline src/Rector/AbstractRector/BetterStandardPrinterTrait.php:107]: Thai should be part of not finde. Printer should only přitom.

Comment (samsonasik) [reply src/Rector/AbstractRector/BetterStandardPrinterTrait.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**